packages/clay-date-picker/src/DateNavigation.js

```diff
@@ -0,0 +1,154 @@
+import * as Helpers from './Helpers';
+import moment from 'moment';
+import PropTypes from 'prop-types';
+import React, {useMemo} from 'react';
+
+import Button from './Button';
+import Select from './Select';
+
+/**
+ * @param {Object} param
+ * @return {React.createElement}
+ */
+function DateNavigation({
+	currentMonth,
+	months,
+	onDotClicked,
+	onMonthChange,
+	spritemap,
+	years,
+}) {
+	const memoizedYears = useMemo(
+		() =>
+			Helpers.range(years).map(elem => {
+				return {
+					label: elem,
+					value: elem,
+				};
+			}),
+		[years.start, years.end]
+	);
+
+	const memoizedMonths = useMemo(
+		() =>
+			months.map((month, index) => {
+				return {
+					label: month,
+					value: index,
+				};
+			}),
+		months
+	);
+
+	/**
+	 * Handles the change of the month from the available
+	 * years in the range
+	 * @param {!number} month
+	 */
+	function handleChangeMonth(month) {
+		const date = moment(currentMonth)
+			.clone()
+			.add(month, 'M')
+			.toDate();
+		const year = date.getFullYear();
+
+		if (memoizedYears.find(elem => elem.value === year)) {
+			onMonthChange(date);
+		}
+	}
+
+	/**
+	 * Handles the previous month button
+	 */
+	function handlePreviousMonthClicked() {
```

Comment:
Out of curiosity, how come you chose to use a function declaration vs an arrow function? For simple functions I tend to prefer arrow functions because it takes up less lines. I don't have strong opinions on in, mostly curious.

something like 
```js
const handlePreviousMonthClicked = () => handleChangeMonth(-1);
```

cc: @wincent what do you think?

Reply:
First answer, the abstract one:

Generally when working with teams I've found that simple rules (because they are easy to understand, easy to apply) are often the best. They tend to lead to the most consistent result, and consistency is super important (even more important than having something "optimal", especially when there is not much difference between the simple-and-consistent version and the "optimal" version).

So in the past, a pattern I've followed is `function` by default, and arrow functions where I want to lock in a `this` value (for example, in event handlers in React components). This pattern is simple enough to explain and follow. In side projects or smaller teams I am much more likely to be nuanced, following a heuristic like the one you're advocating here ("if very simple, turn into a one-liner") which involves an element of judgment and subjectivity.

Second answer, the concrete one:

In this instance I'd probably use an arrow function. 😂  The function is long and using something that is shorter is going to improve the readability of it.

Reply:
🙂 I do not have strong arguments against this haha, but I always set default `function` I think it was more customs. But this case really is simpler arrow function, now other cases I think it is just changing for each other, so I think we should decide if inside hooks we will always use `arrow functions` or just `function` just to maintain a consistency . makes sense?

Reply:
I'm fine either way, we could also just say "use discretion" and not be super firm one way or the other. In the end I would likely favor readability the most.